lib/archive_importer.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+class ArchiveImporter
+  include ArchiveHelper
+  include Diaspora::Logging
+
+  attr_accessor :user
+
+  def initialize(archive_hash)
+    @archive_hash = archive_hash
+  end
+
+  def import
+    import_tag_followings
+    import_aspects
+    import_contacts
+    import_posts
+    import_relayables
+    import_subscriptions
+    import_others_relayables
+  end
+
+  def create_user(username)
+    allowed_keys = %w[
+      email strip_exif show_community_spotlight_in_stream language disable_mail auto_follow_back
+    ]
+    data = convert_keys(archive_hash["user"], allowed_keys)
+    new_password = SecureRandom.hex
```

Comment:
Shouldn't we already make this a parameter to this class and proxy it through in the service?